opencensus/common/transports/async.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2018, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import atexit
+import threading
+import time
+
+from six.moves import queue
+from six.moves import range
+
+from opencensus.common.transports import base
+
+_DEFAULT_GRACE_PERIOD = 5.0  # Seconds
+_DEFAULT_MAX_BATCH_SIZE = 10
+_WAIT_PERIOD = 1.0  # Seconds
+_WORKER_THREAD_NAME = 'opencensus.common.Worker'
+_WORKER_TERMINATOR = object()
+
+
+class _Worker(object):
+	"""A background thread that exports batches of data.
+
+	:type exporter: :class:`~opencensus.trace.exporters.base.Exporter` or 
+					:class:`~opencensus.stats.exporters.base.StatsExporter`
+	:param exporter: Instances of Exporter objects. Defaults to
+					:class:`.PrintExporter`. The rest options are
+					:class:`.ZipkinExporter`, :class:`.StackdriverExporter`,
+					:class:`.LoggingExporter`, :class:`.FileExporter`.
+
+	:type grace_period: float
+	:param grace_period: The amount of time to wait for pending data to
+						 be submitted when the process is shutting down.
+
+	:type max_batch_size: int
+	:param max_batch_size: The maximum number of items to send at a time
+						   in the background thread.
+	"""
+	def __init__(self, exporter, grace_period=_DEFAULT_GRACE_PERIOD,
+				 max_batch_size=_DEFAULT_MAX_BATCH_SIZE):
+		self.exporter = exporter
+		self._grace_period = grace_period
+		self._max_batch_size = max_batch_size
+		self._queue = queue.Queue(0)
+		self._lock = threading.Lock()
+		self._thread = None
+
+	@property
+	def is_alive(self):
+		"""Returns True is the background thread is running."""
+		return self._thread is not None and self._thread.is_alive()
+
+	def _get_items(self):
+		"""Get multiple items from a Queue.
+
+		Gets at least one (blocking) and at most ``max_items`` items
```

Comment:
`max_items` isn't defined anywhere. Did we perhaps mean to use `max_batch_size`?

Reply:
Yes, it was a mistake inside the comment. I'll fix that, thanks !